File: backend/src/main/java/ecrow/backend/business/abstracts/CustomerService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.Customer;
import ecrow.backend.entities.dtos.CustomerAddDto;
import ecrow.backend.entities.dtos.CustomerDepositBalanceDto;
import ecrow.backend.entities.dtos.CustomerBaseUpdateDto;
import ecrow.backend.entities.dtos.SignInDto;

import java.util.List;

public interface CustomerService {
    DataResult<List<Customer>> getAll();
    DataResult<Customer> getById(Integer id);
    DataResult<Customer> getByEmail(String email);
    DataResult<Customer> getByPhoneNumber(String phoneNumber);
    DataResult<Customer> signIn(SignInDto signInDto);
    Result deleteById(Integer id);

    Result add(CustomerAddDto customerAddDto);
    Result updateBase(CustomerBaseUpdateDto customerBaseUpdateDto);
    Result depositBalance(CustomerDepositBalanceDto customerDepositBalanceDto);
}

File: frontend/src/services/CustomerService.js
import axios from "axios";

export default class CustomerService {

    getAll() {
        return axios.get("http://localhost:8080/api/customer/getAll")
    }

    getById(id) {
        return axios.get(`http://localhost:8080/api/customer/getById?id=${id}`)
    }

    getByPhoneNumber(phoneNumber) {
        return axios.get(`http://localhost:8080/api/customer/getByPhoneNumber?phoneNumber=${phoneNumber}`)
    }

    getByEmail(email) {
        return axios.get(`http://localhost:8080/api/customer/getByEmail?email=ad${email}`)
    }

    signIn(values) {
        return axios.post("http://localhost:8080/api/customer/signIn", values)
    }

    deleteById(id) {
        return axios.delete(`http://localhost:8080/api/customer/deleteById?id=${id}`)
    }

    add(values){
        return axios.post("http://localhost:8080/api/customer/add", values)
    }

    updateBalance(values){
        return axios.patch("http://localhost:8080/api/customer/updateBalance", values)
    }

    updateBase(values){
        return axios.patch("http://localhost:8080/api/customer/updateBase", values)
    }
}